packages/core/src/getStateFromPath.tsx

```diff
@@ -0,0 +1,50 @@
+import { InitialState } from './types';
+
+/**
+ * Utility to parse a path string to initial state object accepted by the container.
+ * This is useful for deep linking when we need to handle the incoming URL.
+ *
+ * @param path Path string to parse and convert, e.g. /foo/bar?count=42.
+ */
+export default function getStateFromPath(path: string): InitialState {
+  const parts = path.split('?');
+  const segments = parts[0].split('/').filter(Boolean);
+  const query = parts[1] ? parts[1].split('&') : undefined;
+
+  let result: InitialState | undefined;
+  let current: InitialState | undefined;
+
+  while (segments.length) {
+    const state = {
+      stale: true as true,
+      routes: [{ name: decodeURIComponent(segments[0]) }],
+    };
+
+    if (current) {
+      current.routes[0].state = state;
+    } else {
+      result = state;
+    }
+
+    current = state;
+    segments.shift();
+  }
+
+  if (current == null || result == null) {
+    throw new Error(`Unable to parse path to state: ${path}`);
```

Comment:
I'm not sure if we want to throw an exception here. It's not problem of our code, that given url is broken. Maybe we should fallback to initial state?